pandas/io/parsers.py

```diff
@@ -1400,16 +1400,21 @@ def __init__(self, kwds):
                         "when specifying a multi-index header"
                     )
 
-        # GH 16338
-        elif self.header is not None and not is_integer(self.header):
-            raise ValueError("header must be integer or list of integers")
-
-        # GH 27779
-        elif self.header is not None and self.header < 0:
-            raise ValueError(
-                "Passing negative integer to header is invalid. "
-                "For no header, use header=None instead"
-            )
+        elif self.header:
+            # GH 16338
+            if not is_integer(self.header):
+                raise ValueError("header must be integer or list of integers")
+            # GH 27394
+            elif self.prefix:
+                raise ValueError(
+                    "Argument prefix must be None if argument header is not None"
+                )
```

Comment:
Let's put `prefix` check first.  The validation on `header` can come afterwards.

Reply:
Okay